### PR TITLE
Enable CDN access

### PR DESCRIPTION
**Pull request by jkonecny12**
Support adding certificates for the CDN access by kickstart.

Use the new pykickstart API.

### CODE REVIEW

**Comment by jkonecny12** [comment]
jenkins, test this please

File: anaconda.spec.in [hunk 1]
```diff
@@ -34,7 +34,7 @@ Source0: %{name}-%{version}.tar.bz2
 %define mehver 0.23-1
 %define nmver 1.0.0-6.git20150107
 %define partedver 1.8.1
-%define pykickstartver 1.99.66.19
+%define pykickstartver 1.99.66.19-5
```
Comment (M4rtinK) [inline anaconda.spec.in]: Looks like pykickstart 1.99.66.20-1 got released for RHEL7 in the end, not the expected -5 patch build.

Comment (jkonecny12) [reply anaconda.spec.in]: Fixed. Thanks!

**Comment by jkonecny12** [comment]
Thanks Radek, renamed.

**Comment by jkonecny12** [comment]
**UPDATED:**
* Fix the above.
* Add small fix for proxy check.
* Add tweak for metadata hash check.

**Comment by jkonecny12** [comment]
jenkins, test this please

**Comment by jkonecny12** [comment]
jenkins, test this please

**Comment by jkonecny12** [comment]
**UPDATED:**
* Small test fix

Finally this PR was tested in composer and the unitests were run locally fine. So this is ready to be merged.

**Comment by jkonecny12** [comment]
Please make a review of the newest changes (new commit).

**Pull request merged**